File: server/models/player.js
const mongoose = require('mongoose');

const playerScheme = mongoose.Schema({
    nickname: {
        type: String,
        // exclude all white spaces of the nickname
        trim: true,
    },
    socketID: {
        type: String,
    },
    points: {
        type: Number,
        default: 0,
    },
    playerType: {
        required: true,
        type: String,
    }
});

module.exports = playerScheme;

File: lib/models/player.dart
class Player {
  final String nickname;
  final String socketID;
  final num points;
  final String playerType;

  Player({
    required this.nickname,
    required this.socketID,
    required this.points,
    required this.playerType,
  });

  Map<String, dynamic> toMap() {
    return <String, dynamic>{
      'nickname': nickname,
      'socketID': socketID,
      'points': points,
      'playerType': playerType,
    };
  }

  factory Player.fromMap(Map<String, dynamic> map) {
    return Player(
      nickname: map['nickname'] as String,
      socketID: map['socketID'] as String,
      points: map['points'] as num,
      playerType: map['playerType'] as String,
    );
  }

  Player copyWith({
    String? nickname,
    String? socketID,
    double? points,
    String? playerType,
  }) {
    return Player(
      nickname: nickname ?? this.nickname,
      socketID: socketID ?? this.socketID,
      points: points ?? this.points,
      playerType: playerType ?? this.playerType,
    );
  }
}